### PR TITLE
Implement Ctrl+S or Command+S for save draft

### LINKED ISSUE #1627: 简单实现了文章编辑器按Ctrl+S保存文章，但是应该与更优雅的实现？
### 1. 该问题的重现步骤是什么？

在文章编辑页面（admin/write-post.php）按下键盘的Ctrl+S

### 2. 你期待的结果是什么？实际看到的又是什么？

期待：文章被保存
实际：文章并没有被保存，反而是弹出来浏览器的保存页面框

### 3. 问题出现的环境

- 操作系统版本：Debian 12.0
- Apache/NGINX 版本：NGINX 1.24.0
- 数据库版本： MySQL 5.6.51
- PHP 版本：8.2.6
- Typecho 版本：1.2.1
- 浏览器版本：Chrome 117.0.5938.89

## 我的修复步骤
在write-post.php文件末端加入：
```html
<script>
    document.getElementById('text').addEventListener('keydown',function(e)
    {
        if (83 == e.keyCode && e.ctrlKey)
        { 
            e.stopPropagation();
            e.preventDefault()
            document.getElementById('btn-save').click() //这就是页面上的保存按钮
            return true;
        }
    })
</script>
```

就可以实现在textarea区域内按Ctrl+S时保存文章

显然，这个实现不太优雅（比如全屏编辑模式下按这个组合键提交表单的时候会退出全屏），顺便问一下作者大大是否有更好地实现？

### PR DESCRIPTION
**Pull request by sy-records**
close #1627

### CODE REVIEW

**Comment by MFWT** [comment]
第二次的提交，好像还是用按按钮的方式啊
saveData那个函数可以用吗？

File: admin/src/js/pagedown.js [hunk 1]
```diff
@@ -1767,6 +1767,8 @@ else
         hooks.addNoop("enterFakeFullScreen");
         hooks.addNoop("exitFullScreen");
 
+        hooks.addNoop("saveDraft");
```
Comment (joyqi) [inline admin/src/js/pagedown.js:1770]: 对 pagedown 来说，叫 `save` 比较合适

**Comment by joyqi** [comment]
借这个 pr 做了一些额外的工作，主要是

1. 重构了自动保存的逻辑
2. 用 `fetch` API 重写了文件上传的逻辑，去掉了之前用的几个老旧的库，大概省了200多kb。

**Comment by joyqi** [comment]
Please verify this PR and confirm:

1. The publish, save, and auto-save functions for articles and pages are working properly
2. The attachment upload function is working properly
3. The attachment edit function is working properly

After confirming these points, please click on "Approve", and I will merge it into the dev branch upon approval.

**Comment by sy-records** [comment]
不能删除附件，可能是之前重构修改的问题，获取不到cid值

https://github.com/typecho/typecho/blob/4c8c64c79ef9faf5a8423d653faef3126d274c81/var/Typecho/Request.php#L182-L188

File: var/Widget/Contents/Attachment/Edit.php [hunk 2]
```diff
@@ -26,6 +27,8 @@
  */
 class Edit extends PostEdit implements ActionInterface
 {
+    use EditTrait;
```
Comment (sy-records) [inline var/Widget/Contents/Attachment/Edit.php:30]:
```
Fatal error: Widget\Contents\Post\Edit and Widget\Contents\EditTrait define the same property ($themeCustomFieldsHook) in the composition of Widget\Contents\Post\Edit. However, the definition differs and is considered incompatible. Class was composed in /var/Widget/Contents/Post/Edit.php on line 29
```

**Comment by sy-records** [comment]
LGTM.

**Comment by MFWT** [comment]
LGTM.

**Pull request merged**